src/qutip_qip/operations/gates.py

```diff
@@ -73,7 +79,8 @@ def x_gate(N=None, target=0):
 
     """
     if N is not None:
-        return gate_expand_1toN(x_gate(), N, target)
+        _deprecation_warnings_gate_expansion()
```

Comment:
I'm a bit confused about why we issue the deprecation warning here since we still call `expand_operator` so presumably what this function returns is still the expanded operator? What am  I not understanding?

Reply:
Hi, Simon, thanks for reviewing!

The change here is two-folded. It consists of
- Removing the use of `gate_expand_2toN`, which was deprecated a long time ago.
- Deprecating the expansion of gate unitaries in the gate function. To be consistent with qutip-5.

Reply:
Ah, I see.

Can the expansion be done in a consistent way for more complex gates? E.g. two, three or four qubit gates?

Reply:
Yes, `expand_operator` works for arbitrary qudits gate. It is also included `qutip-v5`.

Reply:
Ah, you mean actually a two-qubit gate with the `dims=[[2,2],[2,2]]`, not `dims=[[4],[4]]`. The current version should also work. There is an example in the function doc, expanding a CNOT gate.